src/classlibnative/bcltype/bignum.h

```diff
@@ -0,0 +1,140 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: bignum.h
+//
+
+//
+
+#ifndef _BIGNUM_H_
+#define _BIGNUM_H_
+
+#include <clrtypes.h>
+
+class BigNum
+{
+public:
+    BigNum();
+    BigNum(UINT32 value);
+    BigNum(UINT64 value);
+    ~BigNum();
+
+    BigNum & operator=(const BigNum &rhs);
+
+    static UINT32 LogBase2(UINT32 val);
+    static UINT32 LogBase2(UINT64 val);
+
+    static int Compare(const BigNum& lhs, UINT32 value);
+    static int Compare(const BigNum& lhs, const BigNum& rhs);
+
+    static void ShiftLeft(UINT64 input, int shift, BigNum& output);
+    static void ShiftLeft(BigNum* pResult, UINT32 shift);
+    static void Pow10(int exp, BigNum& result);
+    static void PrepareHeuristicDivide(BigNum* pDividend, BigNum* divisor);
+    static UINT32 HeuristicDivide(BigNum* pDividend, const BigNum& divisor);
+    static void Multiply(const BigNum& lhs, UINT32 value, BigNum& result);
+    static void Multiply(const BigNum& lhs, const BigNum& rhs, BigNum& result);
+
+    bool IsZero();
+
+    void Multiply(UINT32 value);
+    void Multiply(const BigNum& value);
+    void SetUInt32(UINT32 value);
+    void SetUInt64(UINT64 value);
+    void ExtendBlock(UINT32 newBlock);
+
+private:
+
+    static const UINT8 BIGSIZE = 35;
+    static const UINT8 UINT32POWER10NUM = 8;
+    static const UINT8 BIGPOWER10NUM = 6;
+    static UINT32 m_power10UInt32Table[UINT32POWER10NUM];
```

Comment:
I have to use `constexpr` to accomplish this. The code can be compiled on my current Ubuntu 16.04 box but I'm not sure if it is able to be compiled across platform. I'll do the change and to see if it can pass the CI later.